### PR TITLE
Added logging and code cleanup

### PR DESCRIPTION
**Pull request by Rohesie**
* Adds better logging for unarmed attacks, item attacks and thrown item attacks.

* Includes FF messages for all of them.

* Cleans up some ancient code.

* Fixes thrown items doing zero damage against anything with 1 or more armor.

### CODE REVIEW

**Comment by Rohesie** [comment]
This one should be ready.

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -143,28 +143,25 @@ Contains most of the procs that are called when a mob is attacked by something
...
-		return FALSE
+	else
+		target_zone = def_zone? check_zone(def_zone) : get_zone_with_miss_chance(user.zone_selected, src)
 
 	var/datum/limb/affecting = get_limb(target_zone)
-	if(!affecting)
-		CRASH("[src] called attacked_by [user] with [I] aiming at [def_zone] but found no affecting limb")
 	if(affecting.limb_status & LIMB_DESTROYED)
 		to_chat(user, "What [affecting.display_name]?")
+		log_combat(user, src, "attacked", I, "(FAILED: target limb missing) (INTENT: [uppertext(user.a_intent)]) (DAMTYE: [uppertext(I.damtype)])")
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/human/human_defense.dm:156]: why should this be logged

Comment (Rohesie) [reply code/modules/mob/living/carbon/human/human_defense.dm:156]: Same reason as the other times the attack failed due to missing?

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -143,28 +143,25 @@ Contains most of the procs that are called when a mob is attacked by something
...
-		user.visible_message("<span class='danger'>[user] misses [src] with \the [I]!</span>", null, null, 5)
-		return FALSE
+	else
+		target_zone = def_zone? check_zone(def_zone) : get_zone_with_miss_chance(user.zone_selected, src)
 
 	var/datum/limb/affecting = get_limb(target_zone)
-	if(!affecting)
-		CRASH("[src] called attacked_by [user] with [I] aiming at [def_zone] but found no affecting limb")
 	if(affecting.limb_status & LIMB_DESTROYED)
 		to_chat(user, "What [affecting.display_name]?")
```
Comment (psykzz) [inline code/modules/mob/living/carbon/human/human_defense.dm:155]: What even is this `to_chat`

Comment (spookydonut) [reply code/modules/mob/living/carbon/human/human_defense.dm:155]: it's when you try and hit a missing limb, ala adventure genre games of like "pickup axe" "what axe?"

**Comment by spookydonut** [comment]
@Rohesie i can't update this branch

**Comment by Rohesie** [comment]
I thought allow edits by maintainers was the default option?
It's updated and allowed now, anyway.

**Pull request merged**